driver/src/main/scala/org/mongodb/scala/ObservableImplicits.scala

```diff
@@ -16,420 +16,74 @@
 
 package org.mongodb.scala
 
-import scala.collection.mutable.ListBuffer
-import scala.concurrent.{ExecutionContext, Future, Promise}
-import scala.util.Try
-import com.mongodb.async.client.{Observable => JObservable, Observer => JObserver, Subscription => JSubscription}
+import java.util.concurrent.atomic.AtomicBoolean
+
+import com.mongodb.reactivestreams.client.Success
+import org.mongodb.scala.bson.ObjectId
+import org.mongodb.scala.gridfs.GridFSFile
 import org.mongodb.scala.internal._
+import org.reactivestreams.{Publisher, Subscriber, Subscription => JSubscription}
+
+import scala.concurrent.Future
 
 /**
- * Extends the Java [[Observable]] and adds helpers to make Observables composable and simple to Subscribe to.
+ * Implicit conversion support for Publishers, Observables and Subscriptions
```

Comment:
Most of ObservableImplicits moved into the `Observable` trait.

ObservableImplicits now helps with the translation between raw `Publisher` to scala `Observable`. There is the `SingleObservable` converter as well.

There are also implicit class helpers for type conversion so `Publisher[java.lang.Long]` can become the scala equivalent: `Observable[Long]`

Reply:
Got the walkthrough.  Thanks!